siad/apimisc.go

```diff
@@ -0,0 +1,50 @@
+package main
+
+import (
+	"net/http"
+	"os"
+)
+
+func (d *daemon) updateCheckHandler(w http.ResponseWriter, req *http.Request) {
+	available, version, err := checkForUpdate()
+	if err != nil {
+		http.Error(w, err.Error(), 500)
+		return
+	}
+
+	writeJSON(w, struct {
+		Available bool
+		Version   string
+	}{available, version})
+}
+
+func (d *daemon) updateApplyHandler(w http.ResponseWriter, req *http.Request) {
+	err := applyUpdate(req.FormValue("version"))
+	if err != nil {
+		http.Error(w, err.Error(), 500)
+	}
+
+	writeSuccess(w)
+}
+
+func (d *daemon) statusHandler(w http.ResponseWriter, req *http.Request) {
+	writeJSON(w, d.core.StateInfo())
+}
+
+func (d *daemon) stopHandler(w http.ResponseWriter, req *http.Request) {
+	// TODO: more graceful shutdown?
+	d.core.Close()
+	os.Exit(0)
+}
+
+func (d *daemon) syncHandler(w http.ResponseWriter, req *http.Request) {
+	// TODO: don't spawn multiple CatchUps
+	if len(d.core.AddressBook()) == 0 {
+		http.Error(w, "No peers available for syncing", 500)
+		return
+	}
+
+	go d.core.CatchUp(d.core.RandomPeer())
+
+	writeSuccess(w)
+}
```

Comment:
I'm still of the opinion that this belongs in server/peers. Syncing is about having the same information as your peers. Just like transaction broadcasting and block broadcasting is in the server/peers stuff, syncing should also be there. The job of the server/peers is to make sure everyone is on the same page, and is getting to the same page efficiently.

In my head anyway.